nucypher/policy/conditions/evm.py

```diff
@@ -140,32 +148,49 @@ def chain(self) -> str:
 
     def validate_method(self, method):
         if method not in self.ALLOWED_METHODS:
-            raise Exception(f'{method} is not a permitted RPC endpoint for conditions.')
+            raise ReencryptionCondition.InvalidCondition(
+                f"{method} is not a permitted RPC endpoint for conditions."
+            )
         if not method.startswith('eth_'):
-            raise Exception(f'Only eth RPC methods are accepted for conditions.')
+            raise ReencryptionCondition.InvalidCondition(
+                f"Only eth RPC methods are accepted for conditions."
+            )
         return method
 
     def _configure_provider(self, provider: BaseProvider):
         """Binds the condition's contract function to a blockchian provider for evaluation"""
         self.w3 = Web3(provider)
         provider_chain = self.w3.eth.chain_id
         if provider_chain != self.chain_id:
-            raise Exception(f'This condition can only be evaluated on {self.chain_id} but the providers '
-                            f'connection is to {provider_chain}')
+            raise ReencryptionCondition.InvalidCondition(
+                f"This condition can only be evaluated on {self.chain_id} but the providers "
+                f"connection is to {provider_chain}"
+            )
         return provider
 
+    def _execute_call(self, parameters: List[Any]) -> Any:
+        """Execute onchain read and return result."""
+        rpc_endpoint_, rpc_method = self.method.split("_", 1)
+        web3_py_method = camel_case_to_snake(rpc_method)
+        rpc_function = getattr(
+            self.w3.eth, web3_py_method
+        )  # bind contract function (only exposes the eth API)
```

Comment:
Perhaps you can extract this bit to yet another helper method for further isolation and improved mocking capability.

Reply:
👍